src/lib/table/table-data-source.ts

```diff
@@ -91,14 +91,24 @@ export class MatTableDataSource<T> implements DataSource<T> {
   }
 
   /**
-   * Transforms data objects into a filter term that will be used to check against the filter if
-   * a filter is set. By default, the function will iterate over the values of the data object
-   * and convert them to a lowercase string.
-   * @param data Data object to convert to a string that checked for containing the filter term.
+   * Checks if a data object matches the data source's filter string. By default, each data object
+   * is converted to a string of its properties and returns true if the filter has
+   * at least one occurrence in that string. By default, the filter string has its whitespace
+   * removed and the match is case-insensitive. May be overriden for a custom implementation of
```

Comment:
"...has its whitespace **trimmed** and the match..."?

To me, "removed" kind of implies that `{ name: "Mr. Smith" }` would end up as `mr.smith`

Reply:
Good call, I can see how that could be taken both ways